packages/tools/client-debugger/client-debugger-view/src/components/AudienceView.tsx

```diff
@@ -62,6 +62,51 @@ export function AudienceView(props: AudienceViewProps): React.ReactElement {
 		};
 	}, [clientDebugger, audience, setAllAudienceMembers, setAudienceHistory]);
 
+	const audienceClientMetaData: AudienceClientMetaData[] = [];
+
+	for (const [clientId, client] of allAudienceMembers) {
+		audienceClientMetaData.push({ clientId, client });
+	}
+
+	return (
+		<_AudienceView
+			clientId={myClientId}
+			audienceClientMetaData={audienceClientMetaData}
+			myClientConnection={myClientConnection}
+			onRenderAudienceMember={onRenderAudienceMember}
+			audienceHistory={audienceHistory}
+		/>
+	);
+}
+
+/**
+ * {@link _AudienceView} input props.
+ */
+export interface _AudienceViewProps {
+	clientId: string | undefined;
+	audienceClientMetaData: AudienceClientMetaData[];
+	myClientConnection: IClient | undefined;
```

Comment:
I don't think we need this prop. We can derive this within `_AudienceView` by finding the element in `audienceClientMetadata` that has our `clientId`.

You will probably have to change `MembersView` to take in the `userId` instead of the `clientConnection`, but this seems like an improvement anyways.

Reply:
Now, `myClientConnection` is initialized in `_AudienceView` by `const myClientConnection = audienceClientMetaData.find(foo)`